### PR TITLE
Remove test_helpers, turn on deny(missing_docs) for h3 and h3-quinn

### PR DESCRIPTION
**Pull request by g2p**
This requires all public APIs for both crates to be documented, which involved removing the `test_helpers` feature and making h3 tests part of the h3 crate.

Because of an issue with cyclic dependencies and resolution of the quic traits ([described here](https://www.reddit.com/r/rust/comments/57cb7u/library_crate_that_uses_an_implementation_crate/) and in a comment at the top of tests.rs), tests running with `cfg(test)` access h3_quinn as a submodule of the h3 crate.

### CODE REVIEW

File: h3/src/tests.rs [hunk 1]
```diff
@@ -1,16 +1,28 @@
+// This is to avoid an import loop:
```
Comment (eagr) [inline h3/src/tests.rs:1]: Neat! Out of curiosity, any particular reason this can't be h3/src/tests/mod.rs?

Comment (g2p) [reply h3/src/tests.rs:1]: Sure, I defaulted to the other style but I can change it to match.  Apparently there are clippy lints for this (clippy::self_named_module_files/clippy::mod_module_files).

**Review by eagr — APPROVED** [review]
LGTM @seanmonstar

**Comment by seanmonstar** [comment]
Sorry for the delay in merging (eagr, you're able to merge things when you want too :)), looks great to me too!

**Pull request merged**